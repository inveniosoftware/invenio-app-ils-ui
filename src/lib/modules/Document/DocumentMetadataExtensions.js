import PropTypes from 'prop-types';
import React from 'react';
import { Divider, Table } from 'semantic-ui-react';
import _keys from 'lodash/keys';
import { extensionsConfig } from '@config';

export const DocumentMetadataExtensions = ({ extensions, showDivider }) => {
  return (
    <React.Fragment>
      {showDivider && (
        <Divider horizontal>{extensionsConfig.document.label}</Divider>
      )}
      <Table definition>
        <Table.Body>
          {_keys(extensions).map(key => (
            <Table.Row key={key}>
              <Table.Cell width={4}>
                {extensionsConfig.document.fields[key].label}
              </Table.Cell>
              <Table.Cell>{extensions[key]}</Table.Cell>
            </Table.Row>
          ))}
        </Table.Body>
      </Table>
    </React.Fragment>
  );
};

DocumentMetadataExtensions.propTypes = {
  extensions: PropTypes.object.isRequired,
  showDivider: PropTypes.bool,
};

DocumentMetadataExtensions.defaultProps = {
  showDivider: true,
};
