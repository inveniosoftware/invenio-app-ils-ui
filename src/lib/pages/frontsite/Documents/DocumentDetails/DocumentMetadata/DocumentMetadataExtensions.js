import PropTypes from 'prop-types';
import React from 'react';
import { Divider, Table } from 'semantic-ui-react';
import _forOwn from 'lodash/forOwn';

export const DocumentMetadataExtensions = ({ extensions }) => {
  return _forOwn(extensions, (groupProperties, groupName) => {
    return (
      <React.Fragment key={groupName}>
        <Divider horizontal>{groupName}</Divider>
        <Table definition>
          <Table.Body>
            <Table.Row>
              <Table.Cell>Chris</Table.Cell>
            </Table.Row>
            {_forOwn(groupProperties, (value, key) => (
              <Table.Row key={key}>
                <Table.Cell width={4}>{key}</Table.Cell>
                <Table.Cell>{value}</Table.Cell>
              </Table.Row>
            ))}
          </Table.Body>
        </Table>
      </React.Fragment>
    );
  });
};

DocumentMetadataExtensions.propTypes = {
  extensions: PropTypes.object.isRequired,
};
