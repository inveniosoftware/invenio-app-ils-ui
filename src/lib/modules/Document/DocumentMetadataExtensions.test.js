import React from 'react';
import { mount } from 'enzyme';
import { DocumentMetadataExtensions } from './DocumentMetadataExtensions';

jest.mock('@config/extensionsConfig');

describe('DocumentMetadataExtensions tests', () => {
  let component;
  afterEach(() => {
    if (component) {
      component.unmount();
    }
  });

  const extensions = {
    'fooObject:fooProp': 'fooValue',
    'barObject:barProp': 'barValue',
  };

  it('should render the DocumentMetadataExtensions', () => {
    component = mount(<DocumentMetadataExtensions extensions={extensions} />);
    expect(component).toMatchSnapshot();
    const rows = component.find('DocumentMetadataExtensions').find('TableRow');
    expect(rows).toHaveLength(2);
  });
});
