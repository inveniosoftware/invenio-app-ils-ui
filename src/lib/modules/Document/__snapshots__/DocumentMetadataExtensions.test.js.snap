// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentMetadataExtensions tests should render the DocumentMetadataExtensions 1`] = `
<DocumentMetadataExtensions
  extensions={
    Object {
      "barObject:barProp": "barValue",
      "fooObject:fooProp": "fooValue",
    }
  }
  showDivider={true}
>
  <Divider
    horizontal={true}
  >
    <div
      className="ui horizontal divider"
    >
      document
    </div>
  </Divider>
  <Table
    as="table"
    definition={true}
  >
    <table
      className="ui definition table"
    >
      <TableBody
        as="tbody"
      >
        <tbody
          className=""
        >
          <TableRow
            as="tr"
            cellAs="td"
            key="fooObject:fooProp"
          >
            <tr
              className=""
            >
              <TableCell
                as="td"
                width={4}
              >
                <td
                  className="four wide"
                >
                  Foo
                </td>
              </TableCell>
              <TableCell
                as="td"
              >
                <td
                  className=""
                >
                  fooValue
                </td>
              </TableCell>
            </tr>
          </TableRow>
          <TableRow
            as="tr"
            cellAs="td"
            key="barObject:barProp"
          >
            <tr
              className=""
            >
              <TableCell
                as="td"
                width={4}
              >
                <td
                  className="four wide"
                >
                  Bar
                </td>
              </TableCell>
              <TableCell
                as="td"
              >
                <td
                  className=""
                >
                  barValue
                </td>
              </TableCell>
            </tr>
          </TableRow>
        </tbody>
      </TableBody>
    </table>
  </Table>
</DocumentMetadataExtensions>
`;
